code/obj/machinery/singularity.dm

```diff
@@ -292,6 +297,29 @@ for some reason I brought it back and tried to clean it up a bit and I regret ev
 
 		step(src, dir)
 
+///Returns a 2D vector representing the resultant force acting on the singulo by all gravity wells, scaled by their distance
+/obj/machinery/the_singularity/proc/calc_direction()
+	var/list/total_vector = list(0,0) //if only we had vector primitives...
+	var/turf/singulo_turf = get_turf(src)
+	for_by_tcl(artifact, /obj/machinery/artifact/gravity_well_generator)
```

Comment:
Could potentially be a tracking category in order for ease of expanding this further? e.g. TR_CAT_SINGU_MAGNETS